aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,486 @@
+- HTML Helper alternative: ``Html.BeginForm``
+
+Sample:
+
+.. literalinclude::   forms/sample/final/Views/Demo/RegisterFormOnly.cshtml
+  :language: HTML
+
+The Form Tag Helper above generates the following HTML:
+ 
+.. code-block:: HTML
+
+  <form method="post" action="/Demo/Register">
+    <!-- Input and Submit elements -->
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+   </form>
+  
+The MVC runtime generates the ``action`` attribute value from the Form Tag Helper attributes ``asp-controller`` and ``asp-action``. The Form Tag Helper also generates a hidden `Request Verification Token <http://www.asp.net/mvc/overview/security/xsrfcsrf-prevention-in-aspnet-mvc-and-web-pages>`__ to prevent cross-site request forgery. Protecting a pure HTML Form from cross-site request forgery is very difficult, the Form Tag Helper provides this service for you.
+
+Many of the views in the  *Views/Account* folder (generated when you create a new web app with *Individual User Accounts*)contain the ``asp-route-returnurl`` attribute:
+
+.. code-block:: HTML
+  :emphasize-lines: 2
+  
+  <form asp-controller="Account" asp-action="Login" 
+    asp-route-returnurl="@ViewData["ReturnUrl"]" 
+    method="post" class="form-horizontal" role="form">
+
+:Note: With the built in templates, ``returnUrl`` is only populated automatically when you try to access an authorized resource but are not authenticated or authorized. When you attempt an unauthorized access, the security middleware redirects you to the login page with the ``returnUrl`` set.
+
+Using a named route
+^^^^^^^^^^^^^^^^^^^
+
+The ``asp-route`` Tag Helper attribute can also generate markup for the HTML ``action`` attribute. An app with a :doc:`route </fundamentals/routing>`  named ``register`` could use the following markup for the registration page:
+ 
+.. literalinclude::  forms/sample/final/Views/Demo/RegisterRoute.cshtml 
+  :language: HTML
+  :emphasize-lines: 4
+
+The Input Tag Helper
+---------------------
+ 
+The Input Tag Helper binds an HTML `<input> <https://www.w3.org/wiki/HTML/Elements/input>`__ element to a model property in your razor view.
+
+The Input Tag Helper:
+
+- Generates the ``id`` and ``name`` HTML attributes for the model name specified in the ``asp-for`` attribute.
+- Sets the HTML ``type`` attribute value based on the model type and  `data annotation <https://msdn.microsoft.com/en-us/library/system.componentmodel.dataannotations.aspx>`__ attributes applied to the model property
+- Generates `HTML5 <https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5>`__  validation attributes from `data annotation <https://msdn.microsoft.com/en-us/library/system.componentmodel.dataannotations.aspx>`__ attributes applied to model properties
+- Will not overwrite the HTML ``type`` attribute value when one is specified 
+- HTML Helper alternative: ``Html.EditorFor``
```

Comment:
@dougbu  how is this?

HTML Helper alternative to Input Tag Helper
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

``@Html.TextBoxFor()`` and ``@Html.EditorFor()`` have overlapping features with the Input Tag Helper. The Input Tag Helper will automatically set the ``type`` attribute, ``@Html.TextBoxFor()`` will not. ``@Html.EditorFor()`` handles collections, complex objects and templates, the Input Tag Helper does not. ``@Html.TextBoxFor()``, ``@Html.EditorFor()`` and the Input Tag Helper use lambda expressions and are therefore strongly typed. ``@Html.TextBox()`` and ``@Html.Editor()`` do not accept lambda expressions.